api/src/main/java/org/infinispan/api/collections/reactive/KeyValueEntry.java

```diff
@@ -0,0 +1,20 @@
+package org.infinispan.api.collections.reactive;
+
+public class KeyValueEntry<K, V> {
+   private final K key;
+   private final V value;
+
+   public KeyValueEntry(K key, V value) {
+
```

Comment:
This is basically the same as `org.infinispan.util.KeyValuePair` .. if we're going to use this a lot, we should probably deprecate `KeyValuePair`.

Also, I can see this class being useful outside of just reactive contexts. Maybe we should put in a more generic package, such as `org.infinispan.api.collections.util` or just `org.infinispan.api.collections`, WDYT?